src/servers/src/proto.rs

```diff
@@ -175,8 +176,14 @@ impl PromTimeSeries {
                     return Err(DecodeError::new("delimited length exceeded"));
                 }
                 if label.name.deref() == METRIC_NAME_LABEL_BYTES {
-                    // safety: we expect all labels are UTF-8 encoded strings.
-                    let table_name = unsafe { String::from_utf8_unchecked(label.value.to_vec()) };
+                    let table_name = if is_strict_mode {
+                        match String::from_utf8(label.value.to_vec()) {
```

Comment:
ditto

Reply:
Should we put DecodeError into Snafu for unified handling, as other parts of the function are using DecodeError?

Reply:
> Should we put DecodeError into Snafu for unified handling, as other parts of the function are using DecodeError?

nevermind, it's not a blocking issue.